projects/driver_controls/test/test_pedal_calibration.c

```diff
@@ -0,0 +1,71 @@
+#include "ads1015_def.h"
+#include "debouncer.h"
+#include "delay.h"
+#include "gpio_it.h"
+#include "input_event.h"
+#include "interrupt.h"
+#include "pedal_calibration.h"
+#include "test_helpers.h"
+#include "unity.h"
+
+static PedalCalibrationStorage s_storage;
+static Ads1015Storage s_ads1015_storage;
+static ThrottleCalibrationData s_throttle_calibration_data;
+static DebouncerInfo s_debouncer_info;
+static bool s_continue_flag;
+#define TEST_PEDAL_CALIBRATION_ADC_CHANNEL_A ADS1015_CHANNEL_0
+#define TEST_PEDAL_CALIBRATION_ADC_CHANNEL_B ADS1015_CHANNEL_1
+#define TEST_PEDAL_CALIBRATION_BRAKE_PERCENTAGE 30
+#define TEST_PEDAL_CALIBRATION_COAST_PERCENTAGE 10
+
+static void prv_wait_for_button(void) {
+  s_continue_flag = false;
+  while (!s_continue_flag) {
+  }
+}
+
+static void prv_continue_calibration(const GPIOAddress *address, void *context) {
+  s_continue_flag = true;
+}
+
+void setup_test(void) {
+  gpio_init();
+  interrupt_init();
+  gpio_it_init();
+  soft_timer_init();
+  I2CSettings i2c_settings = {
+    .speed = I2C_SPEED_FAST,                    //
+    .scl = { .port = GPIO_PORT_B, .pin = 10 },  //
+    .sda = { .port = GPIO_PORT_B, .pin = 11 },  //
+  };
+  i2c_init(TEST_ADS1015_I2C_PORT, &i2c_settings);
+  GPIOAddress ready_pin = {
+    .port = GPIO_PORT_B,  //
+    .pin = 2,             //
+  };
+  GPIOAddress calibration_button = {
```

Comment:
I don't think any of our buttons are using raw GPIO - you should just trigger on an event.

Reply:
I can trigger an event but how would I acknowledge the pressing of the button in the first place if it is not GPIO?

Reply:
Trigger on an event. Use an event-driven process instead of busy-waiting on a specific GPIO.